### DISCUSSION TITLE
Triggering button input with gamepad stick

### DISCUSSION THREAD

**Discussion by ndarilek** [Leafwing-Studios/leafwing-input-manager#605] [Q&A] [answered]
Hello,

Converting my game to 0.15 and really like the API clarity. One thing I'm struggling with, though, is how to convert some button-like actions that use gamepad sticks. Here, for instance, is some non-functional code partway ported to 0.15:

```
                    .insert(
                        NavigationAction::SnapCardinal,
                        UserInput::Single(
                            SingleAxis::positive_only(
                                GamepadAxisType::RightStickY,
                                EXTREME_DEADZONE,
                            )
                            .into(),
                        ),
                    )
```

This code is configuring an `InputMap`. `SnapCardinal` is a button action.

Is there any way to configure this action such that flicking the right stick up triggers a button press?

Thanks.

**Comment by alice-i-cecile** [comment, marked as answer]
You want [`GamepadControlDirection`](https://github.com/Leafwing-Studios/leafwing-input-manager/blob/c15d0e60cd130825e486f359f0d45b1a975ea2c3/src/user_input/gamepad.rs#L82) :) I'd link you the docs.rs page, but evidently it's not showing up there because of #606!

**Comment by ndarilek** [comment]
Perfect, thanks!
